### PR TITLE
Cinemaot dataset loader

### PR DESCRIPTION
**Pull request by Lilly-May**
**PR Checklist**

-   [X] Referenced issue is linked (contributes toward #400, but another PR is needed to address the issue completely)
-   [X] Documentation in `docs` is updated

**Description of changes**

I have added the full cinemaot dataset (9209 cells). Previously, only a subsampled version was available through `pt.dt.cinemaot_example()`. However, to provide an example for the `cinemaot.synergy()` method, we need the full dataset because it contains cells with all three stimulations, while the subsampled version contains only one stimulation.

The subsampled dataset is still available via `pt.dt.cinemaot_example()`, the full dataset can now be obtained using `pt.dt.cinemaot_full()`.

### CODE REVIEW

File: pertpy/data/_datasets.py [hunk 1]
```diff
@@ -1429,3 +1432,31 @@ def cinemaot_example() -> AnnData:  # pragma: no cover:
...
+    of the dataset is available via cinemaot_example().
+
+    References:
+        Preprint: https://doi.org/10.1101/2022.07.31.502173
+        Dataset available here: https://datadryad.org/stash/dataset/doi:10.5061/dryad.4xgxd25g1
+
+    Returns:
+        :class:`~anndata.AnnData` object of PBMCs stimulated with interferon.
+    """
+    output_file_name = "Integrated_subset.h5ad"
```
Comment (Zethson) [inline pertpy/data/_datasets.py:1451]: Let's use a better output name here.

Can we use the authors name? We can ask Mingze if it's unclear who generated it...

**Review by Zethson — COMMENTED** [review]
Great!

File: pertpy/data/_datasets.py (hunk 1 above)
Comment (Lilly-May) [reply pertpy/data/_datasets.py:1451]: Based on what is written as a proposed citation [here](https://datadryad.org/stash/dataset/doi:10.5061/dryad.4xgxd25g1), I'd then propose _Dong_2023_?

Comment (Zethson) [reply pertpy/data/_datasets.py:1451]: Yes!

**Review by Zethson — APPROVED** [review]

**Comment by Zethson** [comment]
On a second thought we should rename it to `dong_2023` instead of `cinemaot_full`?

**Pull request merged**